### ISSUE TITLE
How to transfer predicted cell labels from prediction assay to SCT in Visium 10X data

### ISSUE THREAD

**Issue by olhakholod** [satijalab/seurat#8323]
Hello!

I have transferred cell annotations from skin cell atlas to slice of Visium 10X data.
```
# First, we load Seurat and the other packages
library(Seurat)
library(SeuratData)
library(ggplot2)
library(patchwork)
library(dplyr)

# Load 10x Visium dataset
AD_4_LS <- Load10X_Spatial("/Users/f006qpk/Desktop/RO3\ supplement/datasets/spatial/GSE197023\ Visium\ skin/data/AD_4_LS",slice="AD_4_LS")
AD_4_LS$orig.ident <- "AD_4_LS"
skin <- AD_4_LS

# Data preprocessing
skin <- SCTransform(skin, assay = "Spatial", verbose = FALSE)

# Dimensionality reduction, clustering, and visualization
skin <- RunPCA(skin, assay = "SCT", verbose = FALSE)
skin <- FindNeighbors(skin, reduction = "pca", dims = 1:30)
skin <- FindClusters(skin, verbose = FALSE)
skin <- RunUMAP(skin, reduction = "pca", dims = 1:30)

# Identification of Spatially Variable Features
de_markers <- FindMarkers(skin, ident.1 = 0, ident.2 = 2)

skin <- FindSpatiallyVariableFeatures(skin, assay = "SCT", features = VariableFeatures(skin)[1:1000],
                                       selection.method = "moransi")

# Now we visualize the expression of the top 6 features identified by this measure
top.features <- head(SpatiallyVariableFeatures(skin, selection.method = "moransi"), 6)
#SpatialFeaturePlot(skin, features = top.features, ncol = 3, alpha = c(0.1, 1))

# Integration with single-cell data
# Load the reference
reference <- readRDS("/Users/f006qpk/Desktop/seuratobj_HTCA_integrated_by_Project.rds")

# note that setting ncells=3000 normalizes the full dataset but learns noise models on 3k
# cells this speeds up SCTransform dramatically with no loss in performance
library(dplyr)
reference <- SCTransform(reference, ncells = 3000, verbose = FALSE) %>%
  RunPCA(verbose = FALSE) %>%
  RunUMAP(dims = 1:30)

# After subsetting, we renormalize skin
skin <- SCTransform(skin, assay = "Spatial", verbose = FALSE) %>%
  RunPCA(verbose = FALSE)

# the annotation is stored in the 'celltype' column of object metadata
DimPlot(reference, group.by = "Cell_Type", label = TRUE)

# Find transfer anchors between spatial and scRNA-seq datasets
DefaultAssay(reference) <- "integrated"
anchors <- FindTransferAnchors(reference = reference, query = skin, normalization.method = "SCT")

predictions.assay <- TransferData(anchorset = anchors, refdata = reference$Cell_Type, prediction.assay = TRUE,
                                  weight.reduction = skin[["pca"]], dims = 1:30)

skin[["predictions"]] <- predictions.assay

# Now we get prediction scores for each spot for each class
DefaultAssay(skin) <- "predictions"
SpatialFeaturePlot(skin, features = c("Keratinocyte", "Macrophage"), pt.size.factor = 1.6, ncol = 2, crop = TRUE)

# Based on these prediction scores, we can also predict cell types whose location is spatially restricted
skin <- FindSpatiallyVariableFeatures(skin, assay = "predictions", selection.method = "moransi",
                                        features = rownames(skin), r.metric = 5, slot = "data")

top.clusters <- head(SpatiallyVariableFeatures(skin, selection.method = "moransi"), 14)

# pulled from the Seurat source code--requires package RColorBrewer
library(RColorBrewer)
SpatialColors <- colorRampPalette(colors = rev(x = brewer.pal(n = 15, name = "Spectral")))
# Keep all annotations
SpatialFeaturePlot(skin, features = c("B Cell", "Basal Epithelial Cell", "Erythroid Cell",
                                      "Fibroblast", "Keratinocyte", "Lymphatic Endothelial Cell",
                                      "Macrophage", "Melanocyte", "Mesenchymal Cell",
                                      "Neutrophil", "Osteoblast", "Stromal Cell", 
                                      "Vascular Endothelial Cell"), ncol = 3) &
  scale_fill_gradientn(limits=c(0, 1), colours=SpatialColors(n=100))

saveRDS(skin,file="AD_4_LS_spatial_annotated.rds")
```
Now I want to use this .rds file to find cell-cell interactions vis CellChat.

```
# Load the required libraries
library(CellChat)
library(patchwork)
options(stringsAsFactors = FALSE)

# Part I: Data input & processing and initialization of CellChat object
# Load data
# Here we load a Seurat object of 10X Visium mouse cortex data and its associated cell meta data
visium.skin <- readRDS("/Users/f006qpk/Desktop/AD_4_LS_spatial_annotated.rds")

library(Seurat)
# show the image and annotated spots
SpatialDimPlot(visium.skin, label = T, label.size = 3)

# Prepare input data for CelChat analysis
data.input = GetAssayData(visium.skin, slot = "data", assay = "SCT") # normalized data matrix
meta = data.frame(labels = Idents(visium.skin), row.names = names(Idents(visium.skin))) # manually create a dataframe consisting of the cell labels
unique(meta$labels) # check the cell labels
```
However, the last line of the code gives me numerical labels:
```
[1] 7 1 0 2 4 6 3 5
Levels: 0 1 2 3 4 5 6 7
```
I understand that my cell annotations (e.g. T cell) is located in "prediction" assay. Is there a way to transfer these cell annotations to "SCT" assay as a category under metadata.

Thank you for your help!
Olha

**Comment by olhakholod** [comment]
Hello! 

I tried this code and it gives me an error. I think there is a typo in the code. Should I do something like that:

predictions.id <- TransferData(anchorset = anchors, refdata = reference$Cell_Type, prediction.assay = FALSE,
                                  weight.reduction = skin[["pca"]], dims = 1:30)

skin[["SCT"]] <- predictions.id

I tried this code and it gives error that there is no corresponding labels in metadata. I think my objective is to have these cell labels in the SCT data slot, because I will use it later in the CellChat.
